test/groovy/TestsPublishResultsTest.groovy

```diff
@@ -230,4 +235,16 @@ class TestsPublishResultsTest extends BasePiperTest {
 
         stepRule.step.testsPublishResults(script: nullScript, failOnError: true)
     }
+
+    @Test
+    void testPublishUnitTestsWithUpdateResultsDoesNotAllowCommandExecution() throws Exception {
+        def injectString = "' -exec touch {} ; rm -rf / # –"
+        helper.registerAllowedMethod('sh', [String], { String cmd ->
+            assertThat(cmd, containsString(BashUtils.quoteAndEscape(injectString)))
```

Comment:
would be nice to see the quote and escaped string instead of using this again. but then the test needs to be adapted when something changes in the method in `BashUtils`.

Reply:
Yeah, how and that quoting works is tested in BashUtilsTest. This just tests that it is used.